logical/system_config.go

```diff
@@ -0,0 +1,29 @@
+package logical
+
+import "time"
+
+// SystemView exposes system configuration information in a safe way
+// for logical backends to consume
+type SystemView interface {
+
+	// DefaultLeaseTTL returns the default lease TTL set in Vault configuration
+	DefaultLeaseTTL() time.Duration
+
+	// MaxLeaseTTL returns the max lease TTL set in Vault configuration; backend
+	// authors should take care not to issue credentials that last longer than
+	// this value, as Vault will revoke them
+	MaxLeaseTTL() time.Duration
+}
+
+type DefaultSystemView struct {
```

Comment:
I'd make this `StaticSystemView` and just take the values instead of a function. That way we can remove the functions from `Core` and just pass in the configured defaults. This makes it really nice to use for testing as well.